test/dialect/mysql/test_compiler.py

```diff
@@ -1207,3 +1308,103 @@ def test_regexp_replace_flags(self):
             "REGEXP_REPLACE(mytable.myid, CONCAT('(?', %s, ')', %s), %s)",
             checkpositional=("ig", "pattern", "replacement"),
         )
+
+
+class MatchExpressionTest(fixtures.TestBase, AssertsCompiledSQL):
+
+    __dialect__ = mysql.dialect()
+
+    matcheble = table(
+        "user",
+        column("firstname", String),
+        column("lastname", String),
+    )
+
+    def test_match_expression(self):
+        firstname = self.matcheble.c.firstname
+        lastname = self.matcheble.c.lastname
+
+        expr = match_(firstname, lastname, against="John Connor")
+
+        self.assert_compile(
+            expr,
+            "MATCH (user.firstname, user.lastname) AGAINST (%s)",
+            dialect=self.__dialect__,
+        )
+
+        self.assert_compile(
+            expr.in_boolean_mode,
+            "MATCH (user.firstname, user.lastname) AGAINST "
+            "(%s IN BOOLEAN MODE)",
+            dialect=self.__dialect__,
+        )
+
+        self.assert_compile(
+            expr.in_natural_language_mode,
+            "MATCH (user.firstname, user.lastname) AGAINST "
+            "(%s IN NATURAL LANGUAGE MODE)",
+            dialect=self.__dialect__,
+        )
+
+        self.assert_compile(
+            expr.with_query_expansion,
+            "MATCH (user.firstname, user.lastname) AGAINST "
+            "(%s WITH QUERY EXPANSION)",
+            dialect=self.__dialect__,
+        )
+
+        self.assert_compile(
+            expr.in_natural_language_mode.with_query_expansion,
```

Comment:
this is good example, why i used exactly properties. It looks SQLish
How should this code look? something like:
```
expr.in_natural_language_mode().with_query_expansion()
```
There are no function calls in SQL. That's why function calls looks wierd in this case

Reply:
UPD: But `DESC` in SQL is not function call to. And it's OK to use `.desc()`